src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.Editor.cs

```diff
@@ -0,0 +1,272 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.RemoveUnnecessaryImports;
+using Microsoft.CodeAnalysis.Rename;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.MoveType
+{
+    internal abstract partial class AbstractMoveTypeService<TService, TTypeDeclarationSyntax, TNamespaceDeclarationSyntax, TMemberDeclarationSyntax, TCompilationUnitSyntax>
+    {
+        private class Editor
+        {
+            private readonly CancellationToken _cancellationToken;
+            private readonly MoveTypeOptionsResult _moveTypeOptions;
+            private readonly State _state;
+            private readonly TService _service;
+
+            private readonly bool _fromDialog;
+            private readonly bool _makeOuterTypesPartial;
+            private readonly bool _makeTypePartial;
+            private readonly bool _renameFile;
+            private readonly bool _renameType;
```

Comment:
this seems to be a whole lot of different behavior in one type.  Why not have distinct types to handle each type of operation you might perform?